src/lib/array.jl

```diff
@@ -245,14 +265,32 @@ for t in subtypes(AbstractWorkerPool)
 end
 @nograd workers
 
+collect_if_dict(x::Dict) = [x for x in x], collect(keys(x))
```

Comment:
I think this could just be:
```suggestion
collect_if_dict(x::Dict) = collect(values(x)), collect(keys(x))
```